spacy_layout/adapters/azure_adapter.py

```diff
@@ -0,0 +1,212 @@
+import os
+from pathlib import Path
+from typing import Any, Union
+
+from azure.ai.documentintelligence import DocumentIntelligenceClient
+from azure.core.credentials import AzureKeyCredential
+from docling_core.types.doc.base import BoundingBox, CoordOrigin
+from docling_core.types.doc.document import (
+    DoclingDocument,
+    PageItem,
+    TableItem,
+    TextItem,
+)
+from docling_core.types.doc.labels import DocItemLabel
+
+# Import python-dotenv for environment variable management (optional)
+try:
+    from dotenv import load_dotenv
+
+    DOTENV_AVAILABLE = True
+except ImportError:
+    DOTENV_AVAILABLE = False
+
+from .base import BackendAdapter
+
+
+class AzureAdapter(BackendAdapter):
+    """Adapter for Azure AI Document Intelligence."""
+
+    def __init__(self, endpoint: str = None, key: str = None, dotenv_path: str = None):
+        """
+        Initialize the Azure adapter.
+
+        Args:
+            endpoint: Azure Document Intelligence endpoint
+            key: Azure Document Intelligence API key
+            dotenv_path: Path to .env file containing Azure credentials
+        """
+        # Load environment variables from .env file if available
+        if DOTENV_AVAILABLE and dotenv_path is not None:
+            load_dotenv(dotenv_path)
+        elif DOTENV_AVAILABLE:
+            load_dotenv()  # Try to load from default locations
+
+        self.endpoint = endpoint or os.environ.get(
+            "AZURE_DOCUMENT_INTELLIGENCE_ENDPOINT"
+        )
+        self.key = key or os.environ.get("AZURE_DOCUMENT_INTELLIGENCE_KEY")
+
+        if not self.endpoint or not self.key:
+            raise ValueError(
+                "Azure Document Intelligence endpoint and key must be provided or "
+                "set as environment variables (AZURE_DOCUMENT_INTELLIGENCE_ENDPOINT, "
+                "AZURE_DOCUMENT_INTELLIGENCE_KEY) or in a .env file"
+            )
+
+        self.client = DocumentIntelligenceClient(
+            endpoint=self.endpoint, credential=AzureKeyCredential(self.key)
+        )
+
+    def _azure_to_docling_document(self, result) -> DoclingDocument:
+        """
+        Convert Azure Document Intelligence result to DoclingDocument.
+
+        Args:
+            result: Azure Document Intelligence analysis result
+
+        Returns:
+            DoclingDocument: Document in Docling format
+        """
+        # Create an empty DoclingDocument
+        doc = DoclingDocument()
```

Comment:
`name` is a mandatory argument for `DoclingDocument` class init so we'll have a Pydantic validation error here.

Reply:
Thanks. I'm not sure if I'm satisfied with this conversion to DoclingDocument. What alternative approaches should I consider? If you have any further suggestions, please let me know.